### PR TITLE
Remove `brew cask automerge`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Everything is done in the GitHub Action now.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 name: GitHub Actions CI
 on:
-  push:
-    branches: master
-  pull_request: []
+  - push
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml]: Don't want this because it'll duplicate jobs on non-fork branch pushes.

Comment (reitermarkus) [reply .github/workflows/tests.yml]: I'm trying to get the output of

```
curl -H 'Accept: application/vnd.github.antiope-preview+json' https://api.github.com/repos/reitermarkus/brew/commits/automerge/check-runs
```

Comment (MikeMcQuaid) [reply .github/workflows/tests.yml]: Sorry, you've lost me here.

Comment (reitermarkus) [reply .github/workflows/tests.yml]: I'm just experimenting.

If the action doesn't run on `push`, the Check Runs API returns an empty array when I make the request above.

Comment (MikeMcQuaid) [reply .github/workflows/tests.yml]: Gotcha, I see. Cool to experiment just don't merge this change into Homebrew/brew, ta 👍

**Pull request merged**